### ISSUE TITLE
Query returns just ERR if the subquery GROUPs BY both a time interval and a tag

### ISSUE THREAD

**Issue by rossmcdonald** [influxdata/influxdb#7968]
#### Steps to reproduce:

1. Write some data
```
> insert giraffes,name=chrono leaves_consumed=12 1485298440000000000
> insert giraffes,name=graf leaves_consumed=8 1485298440000000000
> insert giraffes,name=chrono leaves_consumed=13 1485298445000000000
> insert giraffes,name=graf leaves_consumed=2 1485298445000000000
> insert giraffes,name=chrono leaves_consumed=15 1485298450000000000
> insert giraffes,name=graf leaves_consumed=4 1485298450000000000
> insert giraffes,name=chrono leaves_consumed=13 1485298455000000000
> insert giraffes,name=graf leaves_consumed=5 1485298455000000000

> select * from giraffes
name: giraffes
time                 leaves_consumed name
----                 --------------- ----
2017-01-24T22:54:00Z 12              chrono
2017-01-24T22:54:00Z 8               graf
2017-01-24T22:54:05Z 13              chrono
2017-01-24T22:54:05Z 2               graf
2017-01-24T22:54:10Z 15              chrono
2017-01-24T22:54:10Z 4               graf
2017-01-24T22:54:15Z 13              chrono
2017-01-24T22:54:15Z 5               graf
```

2. Run a query where the subquery's `GROUP BY` clause includes a time interval and tag
```
> SELECT * FROM (SELECT MEAN("leaves_consumed") FROM "giraffes" WHERE time >= '2017-01-24T22:54:00Z' AND time <= '2017-01-24T22:54:15Z' GROUP BY time(10m),"name")

ERR:
```

#### Expected behavior:

I'd expect to get either this:
```
name: giraffes
tags: name=chrono
time                 mean
----                 ----
2017-01-24T22:50:00Z 13.25

name: giraffes
tags: name=graf
time                 mean
----                 ----
2017-01-24T22:50:00Z 4.75
```

Or this:
```
name: giraffes
time                 mean   name
----                 ----   ----
2017-01-24T22:50:00Z 13.25  chrono
2017-01-24T22:50:00Z 4.75   graf
```

#### Actual behavior:

I get `ERR:`.

Pulled from https://github.com/influxdata/influxdb/issues/7875.

**Comment by jsternberg** [comment]
Fixed via #8051.

**Comment by macckoo** [comment]
Hello,
I am running into a very similar issue, please direct me if I am wrong and I'll open a new issue. 

I have the latest (with the fix merged) v1.2.1  installed and I am: 

SELECTing top(value,<Tag>,5) from (select sum(leaves_consumed) as value from <measurement> where time >= '' and time <= '' group by <Tag>)

In the Influx log (attached below) I get "status=2/INVALIDARGUMENT".

@rossmcdonald @jsternberg is this because 'TOP' is used?

******************************************************INFLUX*************************************************
> SELECT top(value,"name",2) FROM (SELECT SUM("leaves_consumed") as value FROM "giraffes" WHERE time >= '2017-01-24T22:54:00Z' AND time <= '2017-01-24T22:54:15Z' GROUP BY time(10m),"name")

actual behavior:
ERR: 

expected behavior:
Should select the top 2 from the sum, and keep the tags

******************************************************INFLUX*************************************************
As in here without the 'TOP'
> SELECT * FROM (SELECT SUM("leaves_consumed") FROM "giraffes" WHERE time >= '2017-01-24T22:54:00Z' AND time <= '2017-01-24T22:54:15Z' GROUP BY time(10m),"name")
name: giraffes
time                name   sum
----                ----   ---
1485298200000000000 chrono 53
1485298200000000000 graf   19
******************************************************INFLUX*************************************************
Subquery used only as a query seems to work:


******************************************************************LOG****************************************
>influxd[6549]: [I] 2017-03-07T12:01:10Z write udp 127.0.0.1:40532->127.0.0.1:58666: write: connection refused service=subscriber
>influxd[6549]: [I] 2017-03-07T12:01:13Z SELECT top(value, "name", 2) FROM (SELECT sum(leaves_consumed) AS value FROM gpgwp_sprint.autogen.giraffes WHERE time >= '2017-
> influxd[6549]: panic: runtime error: index out of range
> systemd[1]: influxdb.service: Main process exited, code=exited, status=2/INVALIDARGUMENT
>systemd[1]: influxdb.service: Unit entered failed state.
> systemd[1]: influxdb.service: Failed with result 'exit-code'.
> systemd[1]: influxdb.service: Service hold-off time over, scheduling restart.
> systemd[1]: Stopped InfluxDB is an open-source, distributed, time series database.

******************************************************************LOG****************************************

**Comment by macckoo** [comment]
Hello,
I am running into a very similar issue, please tell me if I am wrong and I'll open a new issue. 

-I have the latest (with the fix merged) v1.2.1  installed and I am: 

SELECTing top(value,<Tag>,5) from (select sum(leaves_consumed) as value from <measurement> where time >= '' and time <= '' group by <Tag>)

In the Influx log (attached below) I get "status=2/INVALIDARGUMENT".

@rossmcdonald @jsternberg is this because 'TOP' is used?

******************************************************INFLUX*************************************************
> SELECT top(value,"name",2) FROM (SELECT SUM("leaves_consumed") as value FROM "giraffes" WHERE time >= '2017-01-24T22:54:00Z' AND time <= '2017-01-24T22:54:15Z' GROUP BY time(10m),"name")

actual behavior:
ERR: 

expected behavior:
Should select the top 2 from the sum, and keep the tags.

******************************************************INFLUX*************************************************
As in here: (without the 'TOP')
> SELECT * FROM (SELECT SUM("leaves_consumed") FROM "giraffes" WHERE time >= '2017-01-24T22:54:00Z' AND time <= '2017-01-24T22:54:15Z' GROUP BY time(10m),"name")
name: giraffes
time                name   sum
----                ----   ---
1485298200000000000 chrono 53
1485298200000000000 graf   19
******************************************************INFLUX*************************************************
Subquery as a query seems to work:
> SELECT SUM("leaves_consumed") as value FROM "giraffes" WHERE time >= '2017-01-24T22:54:00Z' AND time <= '2017-01-24T22:54:15Z' GROUP BY time(10m),"name"
name: giraffes
tags: name=chrono
time                value
----                -----
1485298200000000000 53

name: giraffes
tags: name=graf
time                value
----                -----
1485298200000000000 19


******************************************************************LOG****************************************
>influxd[6549]: [I] 2017-03-07T12:01:10Z write udp 127.0.0.1:40532->127.0.0.1:58666: write: connection refused service=subscriber
>influxd[6549]: [I] 2017-03-07T12:01:13Z SELECT top(value, "name", 2) FROM (SELECT sum(leaves_consumed) AS value FROM gpgwp_sprint.autogen.giraffes WHERE time >= '2017-
> influxd[6549]: panic: runtime error: index out of range
> systemd[1]: influxdb.service: Main process exited, code=exited, status=2/INVALIDARGUMENT
>systemd[1]: influxdb.service: Unit entered failed state.
> systemd[1]: influxdb.service: Failed with result 'exit-code'.
> systemd[1]: influxdb.service: Service hold-off time over, scheduling restart.
> systemd[1]: Stopped InfluxDB is an open-source, distributed, time series database.

******************************************************************LOG****************************************
